cypress/e2e/homepage.spec.ts

```diff
@@ -0,0 +1,195 @@
+      // Arrange
+      const UPDATED_HERO_BUTTON_TEXT = "This is changed blah blah"
+      toggleSectionDropdown("Hero section")
+
+      // Act
+      cy.get(HERO_SECTION_HIGHLIGHTS_SELECTOR).check()
+      cy.get(HERO_BUTTON_INPUT_SELECTOR).clear().type(UPDATED_HERO_BUTTON_TEXT)
+      cy.get(HERO_SECTION_DROPDOWN_SELECTOR).check()
+      cy.get(HERO_SECTION_HIGHLIGHTS_SELECTOR).check()
+
+      // Assert
+      cy.get(HERO_BUTTON_INPUT_SELECTOR).should(
+        "have.value",
+        UPDATED_HERO_BUTTON_TEXT
+      )
+    })
+
+    it("should limit users to a single resource component", () => {
+      // Arrange
+      // NOTE: Precondition here is that there are NO existing resources
+      cy.get(ADD_SECTION_DROPDOWN_SELECTOR).select(
+        SECTION_DROPDOWN_OPTIONS.DEFAULT
+      )
+      cy.contains("div", "Add a new section")
+        .find("button")
+        .as("addSectionButton")
+        .should("be.disabled")
+
+      // Act
+      cy.get(ADD_SECTION_DROPDOWN_SELECTOR).select(
+        SECTION_DROPDOWN_OPTIONS.RESOURCES
+      )
+      cy.get("@addSectionButton").click()
+
+      // Assert
+      // NOTE: This is an existing bug on `production`
+      // where adding a resource section doesn't disable.
+      cy.get("@addSectionButton").should("be.disabled")
+    })
+
+    it.skip("should snap preview to that section when clicking on the section in the editor", () => {
+      // NOTE: Not testing this because cypress doesn't support viewports checking
+    })
+
+    it("should rearrange the blocks correctly and the preview should also reflect the new order", () => {
+      // Arrange
+      cy.get(ADD_SECTION_DROPDOWN_SELECTOR).select(
+        SECTION_DROPDOWN_OPTIONS.INFOPIC
+      )
+      const addButtonSelector = getAddSectionButtonSelector()
+      cy.get(addButtonSelector).click()
+
+      // NOTE: Initial order is Infobar -> Infopic
+      cy.get(getHandleSelector()).eq(0).as("first").should("contain", "Infobar")
+      cy.get(getHandleSelector()).eq(1).should("contain", "Infopic")
+      const infobarPreviewSelector = getPreviewSectionSelectorByTitle("Infobar")
+      const infopicPreviewSelector = getPreviewSectionSelectorByTitle("Infopic")
+      cy.get(infobarPreviewSelector)
+        .parent()
+        .next()
+        .should("contain", "Infopic")
+
+      // Act
+      // reorder operation
+      cy.get("@first")
+        .focus()
+        .trigger("keydown", { keyCode: keyCodes.space })
+        // need to re-query for a clone
+        .get("@first")
+        .trigger("keydown", { keyCode: keyCodes.arrowDown, force: true })
+        // finishing before the movement time is fine - but this looks nice
+        .wait(timings.outOfTheWay * 1000)
+        .trigger("keydown", { keyCode: keyCodes.space, force: true })
+
+      // Assert
+      // NOTE: Order is now Infopic -> Infobar
+      // note: not using get aliases as they where returning incorrect results
```

Comment:
Super nit: typo in the comment? should be `will` instead of `where`?

Reply:
typo from the lib spec itself AHHAHA will fix